File: app/src/main/java/com/example/parttimecalander/home/SummationActivity.java
package com.example.parttimecalander.home;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;


import com.example.parttimecalander.R;
import com.example.parttimecalander.home.ui.summationmonth.SummationMonthFragment;


public class SummationActivity extends AppCompatActivity {


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_summation);

        FragmentManager fragmentManager = getSupportFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();

        SummationMonthFragment fragment=new SummationMonthFragment();
        // 프래그먼트를 컨테이너에 교체
        fragmentTransaction.replace(R.id.viewPager, fragment);
        fragmentTransaction.addToBackStack(null); // 뒤로가기 버튼으로 이전 상태로 돌아가기
        fragmentTransaction.commit(); // 변경사항 적용
    }
}

File: app/src/main/java/com/example/parttimecalander/home/SummationActivity.kt
package com.example.parttimecalander.home

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import androidx.viewpager2.adapter.FragmentStateAdapter
import androidx.viewpager2.widget.ViewPager2
import com.example.parttimecalander.home.ui.summationmonth.SummationMonthFragment

import com.example.parttimecalander.R
import com.example.parttimecalander.home.ui.summationmonth.SummationViewModel
import com.example.parttimecalander.home.ui.summationmonth.SummationWeekFragment

class SummationActivity : AppCompatActivity() {

    private lateinit var summationViewModel: SummationViewModel
    private lateinit var viewPager: ViewPager2

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_summation)

        summationViewModel = ViewModelProvider(this).get(summationViewModel::class.java)

        viewPager = findViewById(R.id.viewPager)
        val adapter = ViewPagerAdapter(this)
        viewPager.adapter = adapter
    }
}

//ViewPager 사용해서 month, week fragment를 양옆으로 swipe
//클래스 이름, 생성자 매개변수(이름)
class ViewPagerAdapter(activity: AppCompatActivity) : FragmentStateAdapter(activity){ //fragmentStateAdapter 상속
    override fun getItemCount(): Int = 2 //두 개의 프래그먼트

    override fun createFragment(position: Int): Fragment {
        return when(position){
            0-> SummationMonthFragment()
            else -> SummationWeekFragment()
        }
    }
}